### ISSUE TITLE
Install fails with KeyError: 'processor_architecture'

### ISSUE THREAD

**Issue by fliespl** [pypa/setuptools#626]
Started after yesterdays update

```
Downloading https://pypi.io/packages/source/s/setuptools/setuptools-24.0.0.zip
Extracting in /tmp/tmpmNBMZ1
Now working in /tmp/tmpmNBMZ1/setuptools-24.0.0
Installing Setuptools
Traceback (most recent call last):
  File "setup.py", line 21, in <module>
    exec(init_file.read(), command_ns)
  File "<string>", line 11, in <module>
  File "/tmp/tmpmNBMZ1/setuptools-24.0.0/setuptools/__init__.py", line 14, in <module>
    from setuptools.extension import Extension
  File "/tmp/tmpmNBMZ1/setuptools-24.0.0/setuptools/extension.py", line 11, in <module>
    from . import msvc
  File "/tmp/tmpmNBMZ1/setuptools-24.0.0/setuptools/msvc.py", line 244, in <module>
    class PlatformInfo:
  File "/tmp/tmpmNBMZ1/setuptools-24.0.0/setuptools/msvc.py", line 253, in PlatformInfo
    current_cpu = safe_env['processor_architecture'].lower()
  File "/usr/lib/python2.7/UserDict.py", line 23, in __getitem__
    raise KeyError(key)
KeyError: 'processor_architecture'
Something went wrong during the installation.
See the error message above.
```

**Comment by mattcl** [comment]
I am hitting this as well. I'm pinning to 23.1.0 for now as a workaround. Of interest, it seems to only affect python 2 (we were doing installs of python packages in an ubuntu 14.04 docker container and the python 3 ones worked fine).

**Comment by vadimkry** [comment]
yep. 24.0.0 also getting this

**Comment by anthrotype** [comment]
Could it be related to this? https://github.com/pypa/setuptools/pull/624

**Comment by aabes** [comment]
our workaround was to:
Change L11 of "/usr/local/lib/python2.7/dist-packages/setuptools/extension.py
< from . import msvc9
> from . import msvc9_support

@anthrotype  -  I'd vote for - it's related  ;)

**Comment by JGoutin** [comment]
I hope fixed this.
I anyone with this error want to check it now with the updated version for the PR #627 and report here if fix work. Thanks.

**Comment by JGoutin** [comment]
@boazin I don't manage this project, I am only a contributor.

**Comment by beckmani** [comment]
@JGoutin - I checked your Fix on Linux 14.04.1-Ubuntu with python 2.76.
I'm not getting the exception "KeyError: 'processor_architecture'" but I get the following exception:
Traceback (most recent call last):
  File "/usr/local/bin/easy_install", line 7, in <module>
    from setuptools.command.easy_install import main
  File "/usr/local/lib/python2.7/dist-packages/setuptools/__init__.py", line 14, in <module>
    from setuptools.extension import Extension
  File "/usr/local/lib/python2.7/dist-packages/setuptools/extension.py", line 11, in <module>
    from . import msvc
  File "/usr/local/lib/python2.7/dist-packages/setuptools/msvc.py", line 334, in <module>
    class RegistryInfo:
  File "/usr/local/lib/python2.7/dist-packages/setuptools/msvc.py", line 343, in RegistryInfo
    HKEYS = (winreg.HKEY_USERS,
  File "/usr/lib/python2.7/dist-packages/six.py", line 116, in __getattr__
    _module = self._resolve()
  File "/usr/lib/python2.7/dist-packages/six.py", line 105, in _resolve
    return _import_module(self.mod)
  File "/usr/lib/python2.7/dist-packages/six.py", line 76, in _import_module
    __import__(name)
ImportError: No module named _winreg

**Comment by JGoutin** [comment]
Thanks for reporting.

six seem to not raise correctly Import error on `from setuptools.extern.six.moves import winreg`.
Tried another strategy: disabling this module if not on Windows.

If you want to test.

**Comment by aabes** [comment]
A simpler workaround (while waiting for pypi to be updated) could be to:
export processor_architecture=‘'

prior to attempting to install.

tested on python 2.7.6 (but with somewhat dated get-pip.py)

**Comment by JGoutin** [comment]
@beckmani Thanks for reporting.

Six seem to not correctly raise Import error when importing winreg. I tried another strategy : disabling this module if not on Windows.

Patch updated.

**Comment by beckmani** [comment]
@JGoutin - There is still exception. See line #336
`Traceback (most recent call last):
  File "/usr/local/bin/easy_install", line 7, in <module>
    from setuptools.command.easy_install import main
  File "/usr/local/lib/python2.7/dist-packages/setuptools/__init__.py", line 14, in <module>
    from setuptools.extension import Extension
  File "/usr/local/lib/python2.7/dist-packages/setuptools/extension.py", line 11, in <module>
    from . import msvc
  File "/usr/local/lib/python2.7/dist-packages/setuptools/msvc.py", line 327, in <module>
    class RegistryInfo:
  File "/usr/local/lib/python2.7/dist-packages/setuptools/msvc.py", line 336, in RegistryInfo
    HKEYS = (winreg.HKEY_USERS,
  File "/usr/lib/python2.7/dist-packages/six.py", line 116, in __getattr__
    _module = self._resolve()
  File "/usr/lib/python2.7/dist-packages/six.py", line 105, in _resolve
    return _import_module(self.mod)
  File "/usr/lib/python2.7/dist-packages/six.py", line 76, in _import_module
    __import__(name)
ImportError: No module named _winreg`

**Comment by JGoutin** [comment]
@beckmani Thanks

I updated.

**Comment by JGoutin** [comment]
@beckmani Nice ! Thanks for your tests !

**Comment by jaraco** [comment]
Fixed in #627 and release pending for 24.0.1 release.
